src/control/Edit/Edit.js

```diff
@@ -386,23 +386,23 @@ class Edit extends Control {
       this.snapInteraction = new Snap({
         features: this.layer.getSource().getFeaturesCollection() || new Collection(),
       });
+    }
 
-      // Load all vector layer features in the map and add them to the snap
-      // interaction's feature collection (so they can be snapped to).
-      forEachLayer(this.getMap().getLayerGroup(), (layer) => {
-        if (typeof layer.getSource === 'function') {
-          const source = layer.getSource();
-          if (source !== 'null' && source instanceof VectorSource) {
-            const features = source.getFeatures();
-            if (source.getState() === 'ready' && features.length > 0) {
-              features.forEach((feature) => {
-                this.snapInteraction.addFeature(feature);
-              });
-            }
+    // Load all vector layer features in the map and add them to the snap
+    // interaction's feature collection (so they can be snapped to).
+    forEachLayer(this.getMap().getLayerGroup(), (layer) => {
+      if (typeof layer.getSource === 'function') {
+        const source = layer.getSource();
+        if (source !== 'null' && source instanceof VectorSource) {
+          const features = source.getFeatures();
+          if (source.getState() === 'ready' && features.length > 0) {
+            features.forEach((feature) => {
+              this.snapInteraction.addFeature(feature);
+            });
```

Comment:
I think you're actually fixing another (unreported?/theoretical?) bug with this change...

Without this change, if you started drawing while a vector layer was still loading, it would never get added to the snap interaction.

Now, it will get added if you stop and then start drawing again. In the future we might want to improve it further to also watch for new layers becoming ready and adding them to the snap on the fly.